src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,128 @@
 #include "oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp"
 
 #include "oneapi/mkl/dft/detail/commit_impl.hpp"
+
+#include "dft/backends/mklcpu/mklcpu_helpers.hpp"
 #include "mkl_service.h"
 #include "mkl_dfti.h"
 
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace mklcpu {
+namespace detail {
 
-template <precision prec, domain dom>
-class commit_derived_impl final : public detail::commit_impl {
-public:
-    commit_derived_impl(sycl::queue queue, const detail::dft_values<prec, dom>& config_values)
-            : detail::commit_impl(queue, backend::mklcpu) {
-        DFT_ERROR status = DFT_NOTSET;
-        if (config_values.dimensions.size() == 1) {
-            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom), 1,
-                                          config_values.dimensions[0]);
-        }
-        else {
-            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom),
-                                          config_values.dimensions.size(),
-                                          config_values.dimensions.data());
-        }
-        if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCreateDescriptor failed");
-        }
-
-        set_value(handle, config_values);
+/// Commit impl class specialization for MKLCPU.
+template <dft::detail::precision prec, dft::detail::domain dom>
+class commit_derived_impl : public dft::detail::commit_impl {
+private:
+    static constexpr DFTI_CONFIG_VALUE mklcpu_prec = to_mklcpu(prec);
+    static constexpr DFTI_CONFIG_VALUE mklcpu_dom = to_mklcpu(dom);
+    using mklcpu_desc_t = DFTI_DESCRIPTOR_HANDLE;
 
-        status = DftiCommitDescriptor(handle);
+public:
+    commit_derived_impl(sycl::queue queue, const dft::detail::dft_values<prec, dom> config_values)
```

Comment:
```suggestion
    commit_derived_impl(sycl::queue queue, const dft::detail::dft_values<prec, dom>& config_values)
```
You accidentally resolved this comment

Reply:
addressed this in https://github.com/oneapi-src/oneMKL/commit/1e725f265fb587b92b7786ffc9cb9942bb7b2e7f